src/types/Workload/WorkloadComponent.tsx

```diff
@@ -0,0 +1,123 @@
+import Namespace from '../Namespace';
+import * as React from 'react';
+import { ListViewItem, ListViewIcon, Badge } from 'patternfly-react';
+import { Link } from 'react-router-dom';
+import { PfColors } from '../../components/Pf/PfColors';
+import { IstioLogo } from '../ServiceListComponent';
+
+export const DEPLOYMENT = 'Deployment';
+
+export const dicWorkloadType = {
+  Deployment: 'Deployment'
+};
+
+/** Workload Namespace */
+
+export namespace Workload {
+  export interface WorkloadValidator {
+    renderListItem(index: number): React.ReactElement<{}>;
+    getKind(): string;
+    getLink(): string;
+  }
+
+  export class WorkloadParent {
+    name: string;
+    namespace: Namespace;
+    istioSidecar: boolean;
+    appLabel: boolean;
+    versionLabel: boolean;
+
+    constructor(name: string, namespace: string, istioSidecar: boolean, appLabel: boolean, versionLabel: boolean) {
+      this.name = name;
+      this.namespace = { name: namespace };
+      this.istioSidecar = istioSidecar;
+      this.appLabel = appLabel;
+      this.versionLabel = versionLabel;
+    }
+  }
+
+  export class Deployment extends WorkloadParent implements WorkloadValidator {
+    /** Vars of a Deployment */
+    deploymentIcon = 'process-automation';
+
+    constructor(name: string, namespace: string, istioSidecar: boolean, appLabel: boolean, versionLabel: boolean) {
+      super(name, namespace, istioSidecar, appLabel, versionLabel);
+    }
+
+    getKind = (): string => {
+      return DEPLOYMENT;
+    };
+
+    getLink = (): string => {
+      return '';
```

Comment:
something is missing here? (dependant on another task?)

Reply:
There is not link waiting for workload details page